src/js/components/Form/Form.js

```diff
@@ -165,29 +174,27 @@ const Form = forwardRef(
     ref,
   ) => {
     const { format } = useContext(MessageContext);
-
     const [valueState, setValueState] = useState(valueProp || defaultValue);
     const value = useMemo(
       () => valueProp || valueState,
       [valueProp, valueState],
     );
     const [touched, setTouched] = useState(defaultTouched);
-    const [validationResults, setValidationResults] = useState(
-      defaultValidationResults,
-    );
-    // when onBlur input validation is triggered, we need to complete any
-    // potential click events before running the onBlur validation.
-    // otherwise, click events like reset, etc. may not be registered.
-    // for a detailed scenario/discussion,
-    // see: https://github.com/grommet/grommet/issues/4863
-    // the value of pendingValidation is the name of the FormField
-    // awaiting validation.
-    const [pendingValidation, setPendingValidation] = useState(undefined);
-
+    const [validationResults, setValidationResults] = useState({
+      errors: errorsProp,
+      infos: infosProp,
+    });
+    const validationResultsRef = useRef({});
+    // Simulated onMount state. Consider Form to be mounted once it has
+    // accounted for values originating from controlled inputs (available
+    // at second rendering).
+    const [mounted, setMounted] = useState(false);
     useEffect(() => {
-      setPendingValidation(undefined);
-      setValidationResults({ errors: errorsProp, infos: infosProp });
-    }, [errorsProp, infosProp]);
+      if (!mounted) setMounted('mounting');
```

Comment:
What's are stance on changing types of data? This goes from boolean to string to boolean. This can be hard to understand and have subtle bugs if you have something like `if (!mounted)` .
Would it be better to stick to strings?

Reply:
Good call. I agree. Addressed.